server/service/apple_mdm.go

```diff
@@ -2852,6 +2908,10 @@ func ReconcileAppleProfiles(
 	if err != nil {
 		return ctxerr.Wrap(ctx, err, "getting profiles to install")
 	}
+
+	// Exclude macOS only profiles from iPhones/iPads.
+	toInstall = fleet.FilterMacOSOnlyProfilesFromIOSIPadOS(toInstall)
+
 	toRemove, err := ds.ListMDMAppleProfilesToRemove(ctx)
```

Comment:
> dumb question: since we're filtering only on toInstall, do we need to do the same for toRemove? wouldn't this enqueue RemoveProfile commands for profiles that weren't installed?

It seems that if the profile is not installed, then it's not detected as "toRemove".

> my hunch is that we should really be doing the filtering in the SQL, but I understand that it's not a trivial task, and we need to get this done.

Yeah, I went for the simpler change for now (in Go code).

> another sanity check: do we need to do the same for declarations? For example: Fleet manages a declaration for software updates, but I think that we (Fleet) only support macOS versions there.

Good news around DDMs for software updates is that they apply to hosts member of the macOS Sonoma label, thus it didn't need to be excluded.